tests/test_xmlcomp.py

```diff
@@ -1,133 +1,62 @@
 # -*- coding: utf-8 -*-
 
+from py.path import local
+from pytest_xmlcomp import stringifylist
+from pytest_xmlcomp import XMLJSONFile
+from lxml import etree
+import pytest
 
-def test_bar_fixture(testdir):
-    """Make sure that pytest accepts our fixture."""
+GOOD_DATA = local(__file__).dirpath("data/good")
+BAD_DATA = local(__file__).dirpath("data/bad")
 
-    # create a temporary pytest test module
-    testdir.makepyfile("""
-        def test_sth(bar):
-            assert bar == "europython2015"
-    """)
 
-    # run pytest with the following cmd args
-    result = testdir.runpytest(
-        '--datadir=europython2015',
-        '-v'
-    )
-
-    # fnmatch_lines does an assertion internally
-    result.stdout.fnmatch_lines([
-        '*::test_sth PASSED*',
-    ])
-
-    # make sure that that we get a '0' exit code for the testsuite
-    assert result.ret == 0
-
-
-def test_xmljsonfiles_fixture(testdir):
-    from py.path import local
-    p = local(__file__).dirpath("data")
-    targetdata = testdir.tmpdir.mkdir("data")
-    for f in p.listdir():
+def test_pytest_collect_good_files(testdir):
+    targetdata = testdir.tmpdir.mkdir("gooddata")
+    for f in GOOD_DATA.listdir():
         f.copy(targetdata)
-    testdir.makepyfile("""
-        def test_xmljsonfiles_fixture(xmljsonfiles):
-            assert xmljsonfiles
-        """)
-    result = testdir.runpytest('--datadir=data', '-v')
-    result.stdout.fnmatch_lines([
-        '*::test_xmljsonfiles_fixture PASSED*',
-    ])
+    result = testdir.runpytest()
     assert result.ret == 0
 
-
-def test_xmljsonfiles_noXML_fixture(testdir):
-    from py.path import local
-    p = local(__file__).dirpath("data")
-    targetdata = testdir.tmpdir.mkdir("data")
-    for f in p.listdir():
-        if f.ext == ".xml":
-            continue
+def test_pytest_collect_bad_files(testdir):
+    targetdata = testdir.tmpdir.mkdir("baddata")
+    for f in BAD_DATA.listdir():
         f.copy(targetdata)
-    testdir.makepyfile("""
-        def test_xmljsonfiles_fixture(xmljsonfiles):
-            assert xmljsonfiles
-        """)
-    result = testdir.runpytest('--datadir=data', '-v')
+    result = testdir.runpytest()
     result.stdout.fnmatch_lines([
-        '*::test_xmljsonfiles_fixture FAILED*',
+        "False <class 'list'> [] ['<baz>']"
     ])
     assert result.ret == 1
 
+@pytest.mark.parametrize("test_input, expected", [
+        ([etree.Comment('foobar')], ['<!--foobar-->']),
+        ([etree.ProcessingInstruction("instruction")], ['<?instruction?>']),
```

Comment:
* You can use also `etree.PI()` instead of `etree.ProcessingInstruction()`
* Add another test case with checks the content of the PI. Use `etree.PI('instruction', 'content')`.